runtime/foucoco/src/lib.rs

```diff
@@ -921,6 +924,341 @@ parameter_types! {
+
+				let is_allowed_currency =
+					orml_currencies_allowance_ext::Pallet::<T>::is_allowed_currency(currency_id);
+				if !is_allowed_currency {
+					return Err(DispatchError::Other(
+						"Currency id is not allowed for chain extension",
+					))
+				}
+
+				let result = <orml_currencies::Pallet<T> as MultiCurrency<T::AccountId>>::transfer(
+					currency_id,
+					&address_account,
+					&account_id,
+					balance,
+				);
+
+				warn!("result : {:#?}", result);
+			},
+
+			//balance
+			1106 => {
+				warn!("balance!!!");
+				let mut env = env.buf_in_buf_out();
+				let create_asset: (u8, [u8; 12], [u8; 32], T::AccountId) = env.read_as()?;
+				let (type_id, code, issuer, account_id) = create_asset;
+
+				let currency_id = try_from(type_id, code, issuer).unwrap_or(CurrencyId::Native);
+
+				warn!("asset_id : {:#?}", type_id);
+				warn!("account_id : {:#?}", account_id);
+
+				let is_allowed_currency =
+					orml_currencies_allowance_ext::Pallet::<T>::is_allowed_currency(currency_id);
+				if !is_allowed_currency {
+					warn!("asset_id : {:#?} is_allowed_currency: false", type_id);
+					return Err(DispatchError::Other(
+						"Currency id is not allowed for chain extension",
+					))
+				}
+
+				let balance =
+					<orml_currencies::Pallet<T> as MultiCurrency<T::AccountId>>::free_balance(
+						currency_id,
+						&account_id,
+					);
+
+				warn!("balance : {:#?}", balance);
+
+				env.write(&balance.encode(), false, None)
+					.map_err(|_| DispatchError::Other("ChainExtension failed to call balance"))?;
+			},
+
+			//total_supply
+			1107 => {
+				let mut env = env.buf_in_buf_out();
+				let create_asset: (u8, [u8; 12], [u8; 32], T::AccountId) = env.read_as()?;
+				let (type_id, code, issuer, _account_id) = create_asset;
+
+				let currency_id = try_from(type_id, code, issuer).unwrap_or(CurrencyId::Native);
+
+				let is_allowed_currency =
+					orml_currencies_allowance_ext::Pallet::<T>::is_allowed_currency(currency_id);
+				if !is_allowed_currency {
+					return Err(DispatchError::Other(
+						"Currency id is not allowed for chain extension",
+					))
+				}
+
+				let total_supply =
+					<orml_currencies::Pallet<T> as MultiCurrency<T::AccountId>>::total_issuance(
+						currency_id,
+					);
+
+				env.write(&total_supply.encode(), false, None).map_err(|_| {
+					DispatchError::Other("ChainExtension failed to call total_supply")
+				})?;
+			},
+
+			//approve_transfer
+			1108 => {
+				let ext = env.ext();
+				let address = ext.address().clone();
+				let caller = ext.caller().clone();
+				let mut env = env.buf_in_buf_out();
+				let create_asset: (
+					OriginType,
+					u8,
+					[u8; 12],
+					[u8; 32],
+					T::AccountId,
+					BalanceOfForChainExt<T>,
+				) = env.read_as()?;
+				let (origin_type, type_id, code, issuer, to, amount) = create_asset;
+
+				let from;
+				if origin_type == OriginType::Caller {
+					error!("OriginType::Caller");
```

Comment:
This does not need to be an `error` log does it?

Reply:
Done